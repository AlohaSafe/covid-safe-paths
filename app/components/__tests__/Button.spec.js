import { render } from '@testing-library/react-native';
import React from 'react';

import { Button } from '../Button';

const svg = `<g />`;

it('renders', () => {
  const { asJSON } = render(<Button label='Default' />);

  expect(asJSON()).toMatchSnapshot();
});

it('allows small override', () => {
  const { asJSON } = render(<Button small label='Small' />);

  expect(asJSON()).toMatchSnapshot();
});

it('renders different color for secondary', () => {
  const { asJSON } = render(<Button label='Button' secondary />);

  expect(asJSON()).toMatchSnapshot();
});

it('renders an icon and space-between if supplied', () => {
  const { asJSON } = render(<Button label='Button' icon={svg} />);

  expect(asJSON()).toMatchSnapshot();
});
