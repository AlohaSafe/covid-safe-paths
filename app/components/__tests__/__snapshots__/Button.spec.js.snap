// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows small override 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Small"
    accessibilityRole="button"
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 25,
        "paddingTop": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2e2e2e",
            "fontSize": 17,
            "lineHeight": 28,
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#1f2c9b",
            "fontSize": 19,
            "fontWeight": "500",
          },
        ]
      }
    >
      Small
    </Text>
  </View>
</View>
`;

exports[`renders 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Default"
    accessibilityRole="button"
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 25,
        "paddingTop": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2e2e2e",
            "fontSize": 17,
            "lineHeight": 28,
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#1f2c9b",
            "fontSize": 19,
            "fontWeight": "500",
          },
        ]
      }
    >
      Default
    </Text>
  </View>
</View>
`;

exports[`renders an icon and space-between if supplied 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Button"
    accessibilityRole="button"
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 25,
        "paddingTop": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2e2e2e",
            "fontSize": 17,
            "lineHeight": 28,
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#1f2c9b",
            "fontSize": 19,
            "fontWeight": "500",
          },
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`renders different color for secondary 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Button"
    accessibilityRole="button"
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 25,
        "paddingTop": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2e2e2e",
            "fontSize": 17,
            "lineHeight": 28,
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#1f2c9b",
            "fontSize": 19,
            "fontWeight": "500",
          },
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;
